doc/BUILD.rst

```diff
@@ -0,0 +1,722 @@
+Taisei Project - Build/Maintainer FAQ
+=====================================
+
+This is intended for anyone looking to build or package Taisei Project
+for any of its supported operating systems, including Linux, macOS, Windows,
+Emscripten, Switch, and others.
+
+.. contents::
+
+Dependencies
+------------
+
+Taisei's mandatory dependencies are as follows:
+
+-  C (C11) compiler (``gcc``, ``clang``, etc)
+-  Python >= 3.5
+-  meson >= 0.56.2 (recommended)
+-  ninja >= 1.10.0
+-  docutils
+
+Built-In vs. System Dependencies
+""""""""""""""""""""""""""""""""
+
+Due to the wide array of platforms Taisei supports, the project packages many of
+its core dependencies inside the project using the
+`Meson dependency wrap system <https://mesonbuild.com/Wrap-dependency-system-manual.html>`__.
+This is used to pin specific versions of libraries such as `SDL2` to known-good
+versions, as well as assist with packaging and distribution on older and/or more
+esoteric systems (i.e: Emscripten).
+
+However, recompiling every single dependency Taisei needs every time you want to
+rebuild the project is not ideal. You can still use system dependencies
+installed through your OS's package manager (`apt`, `yum`, `brew`, etc) for
+development, local testing, or simply building the game for your own system.
+
+For convenience, ``meson`` will detect which packages are missing from your
+system and use its wrap dependency system to pull in what it can. The trade-off
+is that this can result in longer build times.
+
+For releases, however, you should rely *exclusively* on the ``meson`` wrap
+dependencies with the Taisei repository, for consistency and
+
+This is controlled through the ``--wrap-mode`` flag with ``meson``. (More
+on that later.)
+
+In summary:
+
+- Development = system-install dependencies are fine (`apt`, `brew`, etc)
+- Release = ``meson`` wrap dependencies only
+
+System Dependencies
+"""""""""""""""""""
+
+Linux
+'''''
+
+(The following example is specific to Ubuntu or Debian, but you can substitute
+`apt` for the package manager for your flavour of Linux.)
+
+The following will install the mandatory tools for building Taisei.
+
+.. code:: sh
+
+   apt update
+   apt install meson cmake build-essential
+
+
+The dependencies below are technically optional, as mentioned above in
+``Built-In Vs. System Dependencies``.
+
+Keep in mind that certain packages may have different names on different
+systems. Ensure you are installing the ``dev`` versions of those packages.
+
+.. code:: sh
+
+   apt install libsdl2-dev libsdl2-mixer-dev libogg-dev libopusfile-dev libpng-dev libzip-dev libx11-dev
+
+If your distribution of Linux uses Wayland as its default window server, ensure
+that Wayland deps are installed:
+
+.. code:: sh
+
+   apt install libwayland-dev
```

Comment:
Honestly I'm not sure about providing explicit package installation instructions for every distro/platform. That means maintaining dependency documentation in multiple places… Which may be challenging as demonstrated by this very example:
* freetype2 is missing
* webp/webpdecoder is missing
* libzstd is missing
* zlib is missing, though there's a very high chance it's already installed if the system has been used to compile literally anything
* docutils are missing
* sdl2-mixer is not needed
* x11 and wayland deps should not be needed, **unless** you're building SDL from a subproject, in which case you'll probably want to install a bunch more dependencies. That's a whole different can of worms; I wouldn't recommend building SDL from the subproject to anyone unless they know exactly what they're doing.